Sources/SwiftGRPCNIO/GRPCHTTPProtocolSwitcher.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+import NIO
+import NIOHTTP1
+import NIOHTTP2
+
+/// Channel handler that creates different processing pipelines depending on whether
+/// the incoming request is HTTP 1 or 2.
+class GRPCHTTPProtocolSwitcher {
+  private let handlersInitializer: ((Channel) -> EventLoopFuture<Void>)
+
+  public init(handlersInitializer: (@escaping (Channel) -> EventLoopFuture<Void>)) {
+    self.handlersInitializer = handlersInitializer
+  }
+}
+
+extension GRPCHTTPProtocolSwitcher: ChannelInboundHandler {
+  typealias InboundIn = ByteBuffer
```

Comment:
This switcher seems to operate on a fairly low level and could be useful for other services supporting both HTTP1 and HTTP2; should we maybe try to get this merged into http://github.com/apple/swift-nio-http2 so others could profit from it, or at least drop the `GRPC` from the class name?

Reply:
Done.

I'd like for this to be provided by `NIOHTTP2`, but I didn't want to wait for it for implementing this. @Lukasa, wdyt?

Reply:
SGTM; leaving open for @Lukasa to comment.